File: java/parser/src/main/java/gov/lbl/parser/parser/Named_argumentsVisitor.java
package gov.lbl.parser.parser;

import gov.lbl.antlr4.visitor.modelicaBaseVisitor;
import gov.lbl.antlr4.visitor.modelicaParser;
import gov.lbl.parser.domain.Named_argument;
import gov.lbl.parser.domain.Named_arguments;

public class Named_argumentsVisitor extends modelicaBaseVisitor<Named_arguments> {
    @Override
    public Named_arguments visitNamed_arguments(modelicaParser.Named_argumentsContext ctx) {
        Named_argumentVisitor named_argumentVisitor = new Named_argumentVisitor();
        Named_argument named_argument = ctx.named_argument() == null ? null : ctx.named_argument().accept(named_argumentVisitor);

        Named_argumentsVisitor named_argumentsVisitor = new Named_argumentsVisitor();
        Named_arguments named_arguments = ctx.named_arguments() == null ? null : ctx.named_arguments().accept(named_argumentsVisitor);

        return new Named_arguments(named_argument, named_arguments);
    }
}

File: jsParser/parser/Named_argumentsVisitor.js
const modelicaVisitor = require('../antlrFiles/modelicaVisitor').modelicaVisitor
const Named_arguments = require('../domain/Named_arguments')

const Named_argumentVisitor = require('./Named_argumentVisitor')

class Named_argumentsVisitor {
  constructor () {
    modelicaVisitor.call(this)
    return this
  }

  visitNamed_arguments (ctx) {
    let named_argument = null
    let named_arguments = null

    if (ctx.named_arguments()) {
      named_arguments = this.visitNamed_arguments(ctx.named_arguments())
    }
    if (ctx.named_argument()) {
      const named_argumentVisitor = new Named_argumentVisitor.Named_argumentVisitor()
      named_argument = named_argumentVisitor.visitNamed_argument(ctx.named_argument())
    }

    return new Named_arguments.Named_arguments(named_argument, named_arguments)
  }
};

Named_argumentsVisitor.prototype = Object.create(modelicaVisitor.prototype)

exports.visitNamed_arguments = this.visitNamed_arguments
exports.Named_argumentsVisitor = Named_argumentsVisitor
